C/common/readingset_circularbuffer.cpp

```diff
@@ -161,9 +135,10 @@ std::vector<std::shared_ptr<ReadingSet>> ReadingSetCircularBuffer::extract(bool
  * @return	Return true if circular buffer is empty otherwise false
  *
  */
-bool ReadingSetCircularBuffer::isBufferEmpty()
+bool ReadingSetCircularBuffer::isEmpty()
 {
-	return m_circularBuffer.empty();
+	lock_guard<mutex> guard(m_mutex);
+	return m_circularBuffer.empty() || (m_nextReadIndex == m_circularBuffer.size());
```

Comment:
This function basically returns whether there are new items that can be read/extracted based on m_nextReadIndex, so either rename this function appropriately or change the logic here.